lib/loader/index.js

```diff
@@ -27,6 +27,9 @@ const KEY_FLOAT = 1 << 20;
 const KEY_NULLABLE = 1 << 21;
 const KEY_MANAGED = 1 << 22;
 
+// TODO: is this the right location for STATICARRAY flag?
+const STATICARRAY = 1 << 23;
```

Comment:
There is [no such flag yet](https://github.com/AssemblyScript/assemblyscript/blob/master/std/assembly/shared/typeinfo.ts#L25). The compiler will have to add it when building RTTI by checking whether a class extends StaticArray.

Reply:
Okay! May I add it to the compiler in this pull request? I just need a bit of guidance for this part.

Reply:
Yeah, without this PR won't work. Relevant code location, besides adding the flag to typeinfo, is: https://github.com/AssemblyScript/assemblyscript/blob/master/src/builtins.ts#L7888. Should be possible to reuse most of what's done for arrays already. Suggesting to move all other flags one up, and adding the static array flag below ARRAY. Must then also update the flag values in the loader accordingly.

Reply:
Okay. Got it.